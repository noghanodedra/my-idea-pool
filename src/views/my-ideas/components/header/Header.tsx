import React from 'react';
import styled from 'styled-components';

const StyledContainer = styled.tr`
  display: flex;
  align-items: center;
  margin-left: 100px;
  margin-bottom: 21px;
  margin-top: 39px;
`;

const StyleHeaderTitle = styled.th`
  height: 20px;
  width: 50px;
  text-align: center;
  margin-left: 34px;
  font-size: 14px;
  font-weight: normal;
  color: #2a3842;
`;
const StyleHeaderTitleNone = styled.th`
  height: 20px;
  width: 460px;
`;

const StyleHeaderTitleAvg = styled(StyleHeaderTitle)`
  margin-left: 45px;
  font-weight: bold;
  font-size: 14px;
  color: #2a3842;
`;

const Header = () => {
  return (
    <StyledContainer>
      <StyleHeaderTitleNone></StyleHeaderTitleNone>
      <StyleHeaderTitle>Impact</StyleHeaderTitle>
      <StyleHeaderTitle>Ease</StyleHeaderTitle>
      <StyleHeaderTitle>Confidence</StyleHeaderTitle>
      <StyleHeaderTitleAvg>Avg.</StyleHeaderTitleAvg>
    </StyledContainer>
  );
};

export default Header;
