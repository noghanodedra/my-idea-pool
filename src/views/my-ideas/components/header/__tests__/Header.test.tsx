import 'jest-styled-components';
import 'mutationobserver-shim';

import { render, screen } from '@testing-library/react';
import { shallow } from 'enzyme';
import React from 'react';

import Header from '../Header';

const renderComponent = () =>
  render(
    <table>
      <thead>
        <Header />
      </thead>
    </table>
  );

describe("Header Component", () => {
  beforeEach(() => {
    renderComponent();
  });

  it("renders correctly", async () => {
    expect(await screen.findAllByText(/Impact/i)).toHaveLength(1);
    expect(await screen.findAllByText(/Avg./i)).toHaveLength(1);
  });

  it("is styled correctly", () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
});
