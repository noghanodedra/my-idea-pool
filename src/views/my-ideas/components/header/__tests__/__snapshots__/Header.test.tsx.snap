// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component is styled correctly 1`] = `
<styled.tr>
  <styled.th />
  <styled.th>
    Impact
  </styled.th>
  <styled.th>
    Ease
  </styled.th>
  <styled.th>
    Confidence
  </styled.th>
  <Styled(styled.th)>
    Avg.
  </Styled(styled.th)>
</styled.tr>
`;
